src/artifactContentRenderers/artifactContentRenderer.ts

```diff
@@ -0,0 +1,59 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ */
+
+import * as vscode from "vscode";
+import * as sarif from "sarif";
+import { BinaryArtifactContentRenderer } from "./binaryArtifactContentRenderer";
+import { TextArtifactContentRenderer } from "./textArtifactContentRenderer";
+import { RenderingArtifactContentRenderer } from "./renderingArtifactContentRenderer";
+
+/**
+ * Interface exposed from artifact content renders.
+ */
+export interface ArtifactContentRenderer {
+    /**
+     * Renders the contents.
+     * @param artifactUri The artifact URI to use as the file-name in the markdown.
+     */
+    render(artifactUri: vscode.Uri): string;
+
+    /**
+     * Computes a range from the content it will render.
+     * @param region The region for the artifact content.
+     */
+    rangeFromRegion?(region: sarif.Region): vscode.Range | undefined;
+
+    /**
+     * Specifies a specific file extension needed for VSCode to detect the
+     * document type. For example, makrdown => '.md'. The extension
+     * should include any necessary periods.
+     */
+    specificUriExtension?: string;
+}
+
+type TryCreateLogFile = (log: sarif.Log, artifactContent: sarif.ArtifactContent, runIndex: number, artifactIndex: number) => ArtifactContentRenderer | undefined;
```

Comment:
>, runIndex: number, artifactIndex: number [](start = 79, length = 41)

Since we are passing the artifact content, these indices are not needed anymore.